src/pages/VenuePage/TemplateWrapper.tsx

```diff
@@ -144,14 +170,27 @@ const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
 
   // @debt remove backButton from Navbar
   return (
-    <ReactionsProvider venueId={venue.id}>
-      <WithNavigationBar fullscreen={fullscreen} hasBackButton={hasBackButton}>
-        <AnnouncementMessage message={venue.bannerMessage} />
-        {template}
-        <ChatSidebar venue={venue} />
-        <UserProfileModal venue={venue} />
-      </WithNavigationBar>
-    </ReactionsProvider>
+    <RelatedVenuesProvider venueId={venue.id}>
+      <ReactionsProvider venueId={venue.id}>
+        <Suspense fallback={<></>}>
+          <WithNavigationBar
+            fullscreen={fullscreen}
+            hasBackButton={hasBackButton}
+          >
+            <Suspense fallback={<></>}>
+              <AnnouncementMessage message={venue.bannerMessage} />
+            </Suspense>
+            {template}
+            <Suspense fallback={<></>}>
+              <ChatSidebar venue={venue} />
+            </Suspense>
+            <Suspense fallback={<></>}>
+              <UserProfileModal venue={venue} />
+            </Suspense>
+          </WithNavigationBar>
+        </Suspense>
+      </ReactionsProvider>
+    </RelatedVenuesProvider>
```

Comment:
As per previous comment, this should probably be reverted due to conflicts with:

- https://github.com/sparkletown/sparkle/pull/1737

Reply:
Since aforementioned PR is merged in staging, b7ec26dc13ee3c39f4d061fe3565ff7cd02b0f3e should bring those component in line with it.